### PR TITLE
Fix UrlsData.get_url

### PR DESCRIPTION
**Pull request by tl-hbk**
### Description

https://github.com/conda/conda/issues/12516

`UrlsData.get_url` was broken for `.conda` packages

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: news/12604-fix-urlsdata-get-url [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* `conda.core.package_cache_data.UrlsData.get_url` no longer fails when `package_path` has `.conda` extension (#12536)
```
Comment (jaimergp) [inline news/12604-fix-urlsdata-get-url:7]:
```suggestion
* `conda.core.package_cache_data.UrlsData.get_url` no longer fails when `package_path` has `.conda` extension (#12516 via #12604)
```

Comment (tl-hbk) [reply news/12604-fix-urlsdata-get-url:7]: Ah whoops my bad was looking at another example of how to do the news file and accidently used their pr/issue number instead

Comment (jaimergp) [reply news/12604-fix-urlsdata-get-url]: No probs, that's what code reviews are for! You have done a great job with the tests and everything! 💪

**Comment by travishathaway** [comment]
@conda-bot

**Comment by travishathaway** [comment]
@tl-hbk,

It looks like we still need you to sign the CLA agreement. Otherwise, everything looks good to me. Thanks for taking the time to add a new test with a descriptive doc string.

**Comment by tl-hbk** [comment]
> @tl-hbk,
> 
> It looks like we still need you to sign the CLA agreement. Otherwise, everything looks good to me. Thanks for taking the time to add a new test with a descriptive doc string.

My legal department had told me that they had completed the CLA for our company and listed me as a contributor.  They had received an acknowledgement email as well

**Comment by travishathaway** [comment]
@conda-bot

**Review by travishathaway — APPROVED** [review]

**Review by sven6002 — APPROVED** [review]
looks good to me.. could run all the tests once before merging plz

**Pull request merged**